capstone/src/main/webapp/script.js

```diff
@@ -1,31 +1,153 @@
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(centerControlDiv);
+
+  //* *button for zipcode data layer */
+  const zipControlDiv = document.createElement('div');
+  const zipControl = new ZipControl(zipControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(zipControlDiv);
+
+  //* *button for zipcode data layer */
+  const precinctControlDiv = document.createElement('div');
+  const precinctControl = new PrecinctControl(precinctControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER]
+      .push(precinctControlDiv);
+}
+
+function CenterControl(controlDiv, map) {
+  //* *button creation and positioning*/
+  const controlUI = document.createElement('div');
+  controlUI.style.backgroundColor = '#fff';
+  controlUI.style.border = '2px solid #fff';
+  controlUI.style.borderRadius = '3px';
+  controlUI.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+  controlUI.style.cursor = 'pointer';
+  controlUI.style.marginBottom = '22px';
+  controlUI.style.textAlign = 'center';
+  controlUI.title = 'Click to recenter the map';
+  controlDiv.appendChild(controlUI);
+
+  //* *css for interior of all buttons*/
+  const text = document.createElement('div');
+  text.style.color = 'rgb(25,25,25)';
+  text.style.fontFamily = 'Roboto,Arial,sans-serif';
+  text.style.fontSize = '16px';
+  text.style.lineHeight = '38px';
+  text.style.paddingLeft = '5px';
+  text.style.paddingRight = '5px';
+  text.innerHTML = 'Center Map';
+  controlUI.appendChild(text);
+
+  //* *button functionality */
+  controlUI.addEventListener('click', function() {
+    map.setCenter({lat: 37.7749, lng: -122.4194});
+  });
+}
+
+let zipClicks = 0;
+function ZipControl(controlDiv, map) {
+  //* *adding zipcode overlay*/
+  const zipcodeLayer = new google.maps.Data({map: map});
+  zipcodeLayer.loadGeoJson('zipcode-data.json');
+  zipcodeLayer.setStyle({visible: false});
+
+  //* *button creation and positioning*/
+  const controlUI = document.createElement('div');
+  controlUI.style.backgroundColor = '#fff';
+  controlUI.style.border = '2px solid #fff';
+  controlUI.style.borderRadius = '3px';
+  controlUI.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+  controlUI.style.cursor = 'pointer';
+  controlUI.style.marginBottom = '22px';
+  controlUI.style.textAlign = 'center';
+  controlUI.title = 'Click to show San Fransisco zip codes';
+  controlDiv.appendChild(controlUI);
+
+  //* *css for interior of all buttons*/
+  const text = document.createElement('div');
+  text.style.color = 'rgb(25,25,25)';
+  text.style.fontFamily = 'Roboto,Arial,sans-serif';
+  text.style.fontSize = '16px';
+  text.style.lineHeight = '38px';
+  text.style.paddingLeft = '5px';
+  text.style.paddingRight = '5px';
+  text.innerHTML = 'Show Zipcodes';
+  controlUI.appendChild(text);
+
+  //* *button functionality */
+  controlUI.addEventListener('click', function() {
+    zipClicks++;
+    if (zipClicks%2==1) {
```

Comment:
Can this be a boolean instead?

Reply:
It could be, but I honestly think that a simple counter will work just the same

Reply:
Personally it was very confusing to read. You are essentially using this as a boolean, so I think it would be a lot clearer if you made it a boolean.

Reply:
I actually like using the mod as opposed to a boolean because visibility is already a boolean and I think that would be more confusing. I can explain mods though to make it clearer: It returns the remainder of dividing the left hand operand by right hand operand.

Reply:
I know what mods do. I just find it confusing and convoluted that you are essentially using the mods to treat the int like a boolean instead of making it a boolean. If you just store the visibility state and change it to !prevState then your code becomes a lot shorter and cleaner. These serve the same purpose so I don't think combining them makes it more confusing.

Reply:
a boolean would certainly suffice here since visibility holds a binary value (true or false/ on or off). I see that you base the visibility on whether the counter is an odd number or not. Personally, I think that is not the clearest way of illustrating the on/off switching of the visibility property but it is all the same.

Reply:
I'm personally leaning towards a boolean for this one as well. Even though both work the same, having a boolean such as `makeZipcodesVisible = true/false` seems to be easier to understand since it's self descriptive.